src/IonTextWriter.ts

```diff
@@ -370,7 +371,11 @@ export class TextWriter implements Writer {
                     this.writeUtf8(s);
                 }
             } else {
```

Comment:
use `else if {` instead of `else { if {`

Think `writeSymbolToken` can be simplified to something like: 
```
if( is_keyword(s) || (isIdentifier(s) && !isSid(s))) {
   // write s as is 
}
else {
   // escape s
}
```

`isSid(s)` function checks for text like `$1` which needs to be escaped to represent `'$1'` instead of the symbol with SID `1`